### ISSUE TITLE
Long icon delay (with easy reproduction steps)

### ISSUE THREAD

**Issue by makusu2** [micheleg/dash-to-dock#1303]
When I open a program, it sometimes takes a few seconds for the icon to show up on the dock.
When I close a program, it sometimes takes a few seconds for the icon to go away.

Reproduction steps: On Ubuntu 20.04, on a new user:
```
gsettings set org.gnome.shell.extensions.dash-to-dock show-mounts false
dconf write /org/gnome/shell/favorite-apps '@as []'
```
Then, log out, and log back in, and you should see the issue after opening/closing a program.
So, it looks like it happens when the dock is completely empty.

single-monitor configuration, Gnome version 3.36.3, I don't know how to get the extension version.

```
➜  ~ dconf dump /org/gnome/shell/extensions/dash-to-dock/
[/]
click-action='minimize'
dash-max-icon-size=16
multi-monitor=false
show-mounts=false
```

**Comment by UA27** [comment]
Sounds like a bug that was fix a few months back by #1222. If you're using the version from extensions.gnome.org, that's probably it. Try using the latest vesion of master from source.

See instructions on readme (really just git clone the repo) -> https://github.com/micheleg/dash-to-dock#installation-from-source

**Comment by makusu2** [comment]
I can no longer reproduce the issue, so I'm going to assume Ubuntu updated their dash-to-dock version. ^_^
